Ll confidence

rinafriedberg opened the pull request:
Added confidence intervals to local linear prediction and overall structure for CI to default prediction strategies.

jtibshirani commented on r-package/grf/bindings/CustomForestBindings.cpp:
```diff
@@ -15,6 +15,7 @@
   along with grf. If not, see <http://www.gnu.org/licenses/>.
  #-------------------------------------------------------------------------------*/
 
+#include <map>
```
I don't think we need to make any changes in this file.

jtibshirani commented on r-package/grf/tests/testthat/test_local_linear_forest.R:
```diff
@@ -115,4 +115,72 @@ test_that("local linear predict returns local linear predictions even without tu
+})
+
+
+test_that("local linear confidence intervals match regression forest with large lambda", {
```
Is this a duplicate of the method above?

rinafriedberg commented:
@jtibshirani These changes should be addressed now!! Let me know if I missed any or if you'd like me to address any other comments. I added M_stored so the value of lambda would correspond to the actual regularization; it looked like otherwise we were just adding each lambda onto the other, because I modified M at each step of the loop.

rinafriedberg commented on core/src/prediction/LocalLinearPredictionStrategy.cpp:
```diff
@@ -52,24 +52,24 @@ std::vector<double> LocalLinearPredictionStrategy::predict(
       size_t current_predictor = linear_correction_variables[j];
-      X(i,j+1) = test_data->get(sampleID, current_predictor)
-              - original_data->get(indices[i],current_predictor);
+      X(i,j+1) = original_data->get(indices[i],current_predictor)
```
Not sure why that showed up -- my branch should read

size_t current_predictor = linear_correction_variables[j];
X(i,j+1) = original_data->get(indices[i],current_predictor)
                 - test_data->get(sampleID, current_predictor);

I've double checked to make sure that's correct in the pushed files now!

rinafriedberg replied:
r.e. Custom forest bindings-  I thought I updated that file to match the current grf implementation exactly and erase any changes I made before; what changes are showing up?

jtibshirani commented:
@swager committed some tests, that uncovered a discrepancy between the output of `tune_local_linear_forest` and `predict.local_linear_forest`. I fixed the issue on `master` in #324 and will merge it in.
